order.go

```diff
@@ -154,38 +155,49 @@ type Address struct {
 	Zip          string  `json:"zip,omitempty"`
 }
 
+type DiscountApplication struct {
```

Comment:
There are 3 types of DiscountApplications https://help.shopify.com/en/api/reference/orders/order#discount-applications-property the fields in this struct only cover the fields for one type "discount_code". I'm curious about the reasons you choose to only support one type.